diff --git a/fs/proc/base.c b/fs/proc/base.c
index b6959f6da..660056a49 100644
--- a/fs/proc/base.c
+++ b/fs/proc/base.c
@@ -1012,8 +1012,11 @@ static ssize_t environ_read(struct file *file, char __user *buf,
 	return ret;
 }
 
+extern ssize_t fmac_environ_write(struct file *, const char __user *, size_t, loff_t *);
+
 static const struct file_operations proc_environ_operations = {
 	.open		= environ_open,
+	.write		= fmac_environ_write,
 	.read		= environ_read,
 	.llseek		= generic_file_llseek,
 	.release	= mem_release,
@@ -3148,7 +3151,7 @@ static const struct pid_entry tgid_base_stuff[] = {
 #ifdef CONFIG_NET
 	DIR("net",        S_IRUGO|S_IXUGO, proc_net_inode_operations, proc_net_operations),
 #endif
-	REG("environ",    S_IRUSR, proc_environ_operations),
+	REG("environ", S_IRUSR | S_IWUSR, proc_environ_operations),
 	REG("auxv",       S_IRUSR, proc_auxv_operations),
 	ONE("status",     S_IRUGO, proc_pid_status),
 	ONE("personality", S_IRUSR, proc_pid_personality),
@@ -3566,7 +3569,7 @@ static const struct pid_entry tid_base_stuff[] = {
 #ifdef CONFIG_NET
 	DIR("net",        S_IRUGO|S_IXUGO, proc_net_inode_operations, proc_net_operations),
 #endif
-	REG("environ",   S_IRUSR, proc_environ_operations),
+	REG("environ", S_IRUSR | S_IWUSR, proc_environ_operations),
 	REG("auxv",      S_IRUSR, proc_auxv_operations),
 	ONE("status",    S_IRUGO, proc_pid_status),
 	ONE("personality", S_IRUSR, proc_pid_personality),
