diff --git a/drivers/Kconfig b/drivers/Kconfig
index 9480d8465..43f9eaf5e 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -210,4 +210,6 @@ source "drivers/sensors/Kconfig"
 
 source "drivers/tee/Kconfig"
 
+source "drivers/FMAC/Kconfig"
+
 endmenu
diff --git a/drivers/Makefile b/drivers/Makefile
index 06e2bb462..692e3bdc2 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -179,3 +179,4 @@ obj-$(CONFIG_ESOC)              += esoc/
 obj-$(CONFIG_FPGA)		+= fpga/
 obj-$(CONFIG_SENSORS_SSC)		+= sensors/
 obj-$(CONFIG_TEE)		+= tee/
+obj-$(CONFIG_FMAC)		+= FMAC/
